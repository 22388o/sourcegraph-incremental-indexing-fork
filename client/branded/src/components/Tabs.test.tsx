import React from 'react'
import { TabsWithLocalStorageViewStatePersistence } from './Tabs'
import { mount } from 'enzyme'

describe('Tab', () => {
    test('Tabs with local storage persistenc', () => {
        expect(
            mount(
                <TabsWithLocalStorageViewStatePersistence
                    tabs={[{ id: 'files', label: 'Files' }]}
                    storageKey="repo-revision-sidebar-last-tab"
                    tabBarEndFragment={
                        <>
                            <div>fragment</div>
                        </>
                    }
                    id="tab"
                    className="tab-bar"
                    tabClassName="tab-bar__tab--h5like"
                    onSelectTab={() => null}
                >
                    <div>Panel 1</div>
                    <div>Panel 2</div>
                </TabsWithLocalStorageViewStatePersistence>
            )
        ).toMatchSnapshot()
    })
})
