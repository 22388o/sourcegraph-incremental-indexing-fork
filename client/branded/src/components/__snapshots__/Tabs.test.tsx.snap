// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Tabs with local storage persistenc 1`] = `
<TabsWithLocalStorageViewStatePersistence
  className="tab-bar"
  id="tab"
  onSelectTab={[Function]}
  storageKey="repo-revision-sidebar-last-tab"
  tabBarEndFragment={
    <React.Fragment>
      <div>
        fragment
      </div>
    </React.Fragment>
  }
  tabClassName="tab-bar__tab--h5like"
  tabs={
    Array [
      Object {
        "id": "files",
        "label": "Files",
      },
    ]
  }
>
  <Tabs
    activeTab="files"
    className="tab-bar"
    id="tab"
    onSelectTab={[Function]}
    storageKey="repo-revision-sidebar-last-tab"
    tabBarEndFragment={
      <React.Fragment>
        <div>
          fragment
        </div>
      </React.Fragment>
    }
    tabClassName="tab-bar__tab--h5like"
    tabComponent={[Function]}
    tabs={
      Array [
        Object {
          "id": "files",
          "label": "Files",
        },
      ]
    }
  >
    <div
      className="tabs tab-bar"
      id="tab"
    >
      <TabBar
        activeTab="files"
        endFragment={
          <React.Fragment>
            <div>
              fragment
            </div>
          </React.Fragment>
        }
        tabClassName="tab-bar__tab--h5like"
        tabComponent={[Function]}
        tabs={
          Array [
            Object {
              "id": "files",
              "label": "Files",
            },
          ]
        }
      >
        <div
          className="tab-bar "
          role="tablist"
        >
          <Component
            className="btn btn-link btn-sm tab-bar__tab tab-bar__tab--active tab-bar__tab--h5like"
            key="files"
            role="tab"
            selected={true}
            tab={
              Object {
                "id": "files",
                "label": "Files",
              }
            }
            tabIndex={0}
          >
            <button
              aria-controls="files"
              aria-selected={true}
              className="btn btn-link btn-sm tab-bar__tab tab-bar__tab--active tab-bar__tab--h5like"
              data-test-tab="files"
              onClick={[Function]}
              role="tab"
              type="button"
            >
              Files
            </button>
          </Component>
          <div>
            fragment
          </div>
        </div>
      </TabBar>
    </div>
  </Tabs>
</TabsWithLocalStorageViewStatePersistence>
`;
